dist/pythonlibs/riotctrl_shell/cord_ep.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (C) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+"""
+CORD EP-related shell interactions
+
+Defines CORD EP-related shell command interactions
+"""
+
+import re
+
+from riotctrl.shell import ShellInteraction, ShellInteractionParser
+
+
+# ==== Parsers ====
+
+class CordEpRegistrationInfoParser(ShellInteractionParser):
+
+    def __init__(self):
+        self.comps = {
+            "rdaddr": re.compile(r"RD address:\s+"
+                                 r"(?P<rdaddr>coaps?://"
+                                 r"\[[0-9a-f:]+(%\S+)?\](:\d+)?)"),
+            "epname": re.compile(r"ep name:\s+(?P<epname>.+)$"),
+            "ltime": re.compile(r"lifetime:\s+(?P<ltime>\d+)s$"),
+            "regif": re.compile(r"reg if:\s+(?P<regif>\S+)$"),
+            "location": re.compile(r"location:\s+(?P<location>\S+)$"),
+        }
+
+    def parse(self, cmd_output):
+        """
+        Parses output of CordEp::cord_ep_info() and CordEp::cord_ep_register()
+
+        >>> parser = CordEpRegistrationInfoParser()
+        >>> res = parser.parse("CoAP RD connection status:\\n"
+        ...                "RD address:"
+        ...                " coap://[fe80::48d5:9eff:fe98:6b74]:5683\\n"
+        ...                "   ep name: RIOT-760D2323760D2323\\n"
+        ...                "  lifetime: 60s\\n"
+        ...                "    reg if: /resourcedirectory\\n"
+        ...                "  location: /reg/1/")
```

Comment:
Can you please add a second test case for when there is an interface specified in the IPv6 address? And maybe also the `coaps` case. So mostly the same, just having e.g. `coaps://[fe80::48d5:9eff:fe98:6b74%iface0]:5683` as `RD address`

Reply:
Added test case. Note that the current `cord_ep` shell command hardcodes `coap` into the URL and does not add iface to the address. I'll make a PR to fix this.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/14715#pullrequestreview-462423618